lib/aggregate.js

```diff
@@ -1049,7 +1049,7 @@ Aggregate.prototype.exec = async function exec() {
   const options = clone(this.options || {});
   let result;
   try {
-    const cursor = collection.aggregate(this._pipeline, options);
+    const cursor = await collection.aggregate(this._pipeline, options);
```

Comment:
It looks like this still throws a `MongoNotConnectedError: Client must be connected before running operations`. I'll merge this PR into master so I can more easily debug.